pkg/relayer/proxy/relay_builders.go

```diff
@@ -10,42 +10,45 @@ import (
 
 // newRelayRequest builds a RelayRequest from an http.Request.
 func (sync *synchronousRPCServer) newRelayRequest(request *http.Request) (*types.RelayRequest, error) {
-	requestBz, err := io.ReadAll(request.Body)
+	requestBody, err := io.ReadAll(request.Body)
 	if err != nil {
 		return nil, err
 	}
+	defer request.Body.Close()
```

Comment:
@red-0ne Claude 3 mentioned `Body.Close()` never happens, and I also couldn't find that in the code, so I added it here. Could it be a reason for a goroutine leak? Though the tests seem to pass, I want to check in with you to make sure we're not closing the body twice.

Reply:
It seems dangerous to me to to a `.Close` inside of a helper where the request gets passed in.

I feel like we should do it in `ServeHTTP` inside of `pkg/appgateserver/server.go`.